### PR TITLE
chore(override): enable passing down COPILOT env vars to CDK files

### PR DESCRIPTION
**Pull request by efekarakus**
Related #4208 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/template/templates/overrides/cdk/stack.ts [hunk 1]
```diff
@@ -4,14 +4,19 @@ import * as path from 'path';
...
 {{- end }}
 
+interface TransformedStackProps extends cdk.StackProps {
+    readonly appName?: string;
+    readonly envName?: string;
+}
+
 export class TransformedStack extends cdk.Stack {
-    constructor (scope: cdk.App, id: string, props?: cdk.StackProps) {
+    constructor (scope: cdk.App, id: string, props: TransformedStackProps) {
```
Comment (iamhopaul123) [inline internal/pkg/template/templates/overrides/cdk/stack.ts]: if `appName` and `envName` are optional, why do we make `props` required?

File: internal/pkg/override/cdk.go [hunk 2]
```diff
@@ -16,30 +16,70 @@ import (
...
 	rootAbsPath string // Absolute path to the overrides/ directory.
 
-	out  io.Writer  // Writer for any os/exec calls.
-	fs   afero.Fs   // OS file system.
-	exec Executable // For testing os/exec calls.
+	out  io.Writer // Writer for any os/exec calls.
+	fs   afero.Fs  // OS file system.
+	exec struct {
+		LookPath func(file string) (string, error)
+		Command  func(name string, args ...string) *exec.Cmd
```
Comment (iamhopaul123) [inline internal/pkg/override/cdk.go:27]: Will this be a problem for mock for unit test?

**Review by iamhopaul123 — APPROVED** [review]
LGTM overall :shipit:

File: internal/pkg/template/templates/overrides/cdk/stack.ts (hunk 1 above)
Comment (efekarakus) [reply internal/pkg/template/templates/overrides/cdk/stack.ts]: good question 😂 changed to `appName` and `envName` being required

File: internal/pkg/override/cdk.go (hunk 2 above)
Comment (efekarakus) [reply internal/pkg/override/cdk.go:27]: Nope! In the unit tests we use the public constructor to define our own custom behavior:
```
WithCDK("some/path", CDKOpts{
   LookPathFn: func(string) (string, error) { return "", errors.New("some error") }
})
```

**Review by huanjani — APPROVED** [review]
LGTM!

**Pull request merged**